tsdb/tsm1/engine_test.go

```diff
@@ -307,6 +309,56 @@ func BenchmarkEngine_WritePoints_Parallel(b *testing.B) {
 	}
 }
 
+func BenchmarkEngine_DeletePrefixRange_Cache(b *testing.B) {
+	config := tsm1.NewConfig()
+	config.Cache.SnapshotMemorySize = toml.Size(256 * 1024 * 1024)
+	e, err := NewEngine(config)
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	if err := e.Open(context.Background()); err != nil {
+		b.Fatal(err)
+	}
+
+	pp := make([]models.Point, 0, 100000)
+	for i := 0; i < 100000; i++ {
+		p := MustParsePointString(fmt.Sprintf("cpu-%d,host=%d value=1.2", i%1000, i), fmt.Sprintf("000000001122111100000000112211%d", i%1000))
+		pp = append(pp, p)
+	}
+
+	b.Run("exists", func(b *testing.B) {
+		b.ReportAllocs()
+		for i := 0; i < b.N; i++ {
+			b.StopTimer()
+			if err = e.WritePoints(pp); err != nil {
+				b.Fatal(err)
+			}
+			b.StartTimer()
+
+			if err := e.DeletePrefixRange(context.Background(), []byte("0000000011221111000000001122112"), 0, math.MaxInt64, nil); err != nil {
+				b.Fatal(err)
+			}
+		}
+	})
+
+	b.Run("not exists", func(b *testing.B) {
+		b.ReportAllocs()
+		for i := 0; i < b.N; i++ {
+			b.StopTimer()
+			if err = e.WritePoints(pp); err != nil {
+				b.Fatal(err)
+			}
+			b.StartTimer()
+
+			if err := e.DeletePrefixRange(context.Background(), []byte("fooasdasdasdasdasd"), 0, math.MaxInt64, nil); err != nil {
+				b.Fatal(err)
+			}
+		}
+	})
```

Comment:
This is a pretty minor nit but I personally find it easier for the subtest name to be camel case (or at least use underscores instead of spaces) since `go test` will replace the spaces with underscores. e.g. `t.Run("NotExists")` or `t.Run("not_exists")`. That way it's easier to search for if you just have the test name.